libs/database/src/pg_row.rs

```diff
@@ -275,3 +275,93 @@ impl From<AFReactionRow> for Reaction {
     }
   }
 }
+
+#[derive(Debug, FromRow, Serialize)]
+pub struct AFTemplateCategoryRow {
+  pub id: Uuid,
+  pub name: String,
+  pub icon: String,
+  pub bg_color: String,
+  pub description: String,
+  pub category_type: AFTemplateCategoryTypeColumn,
+  pub priority: i32,
+}
+
+impl From<AFTemplateCategoryRow> for TemplateCategory {
+  fn from(value: AFTemplateCategoryRow) -> Self {
+    Self {
+      id: value.id,
+      name: value.name,
+      icon: value.icon,
+      bg_color: value.bg_color,
+      description: value.description,
+      category_type: value.category_type.into(),
```

Comment:
is it not possible to represent `category_type` from returned row in sqlx? i am thinking if the type conversion boilerplates can be avoided

Reply:
Currently, the category type is being stored as int type in postgres. So one way to avoid such conversion would be to use integer type for category_type field as opposed to enum - though I would prefer not to in this case, since we only have a limited number of category type.

As far as i know it is not possible to automatically convert an integer to enum type without having a struct that derives sqlx type.